src/base/Context.ts

```diff
@@ -242,93 +198,94 @@ export class Context {
-      throw new PolymeshError({
-        code: ErrorCode.General,
-        message: 'There is no Account associated with the SDK',
-      });
-    }
+  public async setSigningManager(signingManager: SigningManager): Promise<void> {
+    this._signingManager = signingManager;
 
-    const pairs = [...keyring.getPairs()];
+    signingManager.setSs58Format(this.ss58Format.toNumber());
 
-    const [first] = remove(pairs, ({ address }) => currentPair.address === address);
+    // this could be undefined
+    const [firstAccount] = await signingManager.getAccounts();
 
-    return [
-      new Account({ address: first.address }, this),
-      ...pairs.map(({ address }) => new Account({ address }, this)),
-    ];
+    if (!firstAccount) {
+      this.signingAddress = undefined;
+    } else {
+      return this.setSigningAddress(firstAccount);
+    }
   }
 
   /**
    * @hidden
-   *
-   * Add a signing pair to the Keyring
    */
-  public addPair(params: AddPairParams): KeyringPair {
-    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-    return Context._addPair({ ...params, keyring: this.keyring })!;
+  private get signingManager(): SigningManager {
+    const { _signingManager: manager } = this;
+
+    if (!manager) {
+      throw new PolymeshError({
+        code: ErrorCode.General,
+        message: 'There is no Signing Manager attached to the SDK',
+      });
+    }
+
+    return manager;
   }
 
   /**
    * @hidden
+   *
+   * Retrieve a list of Accounts that can sign transactions. The first Account in the array is the current (default) Account
    */
-  private static _addPair(params: AddPairBaseParams & AddPairParams): KeyringPair | undefined {
-    const { accountSeed, accountUri, accountMnemonic, keyring, pair } = params;
-
-    let newPair: KeyringPair;
-    if (accountSeed) {
-      if (accountSeed.length !== 66) {
-        throw new PolymeshError({
-          code: ErrorCode.ValidationError,
-          message: 'Seed must be 66 characters in length',
-        });
-      }
+  public async getSigningAccounts(): Promise<Account[]> {
+    const { signingManager } = this;
 
-      newPair = keyring.addFromSeed(hexToU8a(accountSeed), undefined, 'sr25519');
-    } else if (accountUri) {
-      newPair = keyring.addFromUri(accountUri);
-    } else if (accountMnemonic) {
-      newPair = keyring.addFromMnemonic(accountMnemonic);
-    } else if (pair) {
-      /*
-       * NOTE @monitz87: the only way to avoid this assertion is to import the Keyring package
-       *   which doesn't make sense just for this
-       */
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      newPair = keyring.addPair(pair as any);
-    } else {
-      return;
-    }
+    let accounts = await signingManager.getAccounts();
+
+    /*
+     * we clone the array so as to not mess with the Signing Manager's internal representation
+     * when we remove the first element
+     */
+    accounts = tuple(...accounts);
```

Comment:
```suggestion
    accounts = [...accounts];
```
I think its easier to read then wondering what tuple is and having to parse the generic stuff its doing.

An alternative would be to use filter + reduce, and set the signing account as the initial address.

```ts

const makeAccount = (address) => new Account({address}, this)

return accounts
  .filter(a => a !== signingAddress)
  .reduce( (acc, address) => acc.push(makeAccount(address)), [makeAccount(signingAddress])
```

(not tested, but I prefer not mutating state when possible)

Reply:
Agreed. I was using `tuple` because of a typing issue back when `accounts` was `string[] | IKeyringPair[]`. Right now this is not necessary